### PR TITLE
Get the real repository source to update the generator

### PR DESCRIPTION
**Pull request by landru29**
In my company, we are using yo. I developped internal generators (only valid for my company, that's why I cannot share on github)
With this modification, when updating, the exact source of the repository (if specified) is used to update the package

### CODE REVIEW

**Comment by landru29** [comment]
Why all tests are crashing with node6 ?!

File: lib/routes/update.js [hunk 1]
```diff
@@ -16,8 +16,22 @@ function updateSuccess(app) {
   });
 }
 
+function getPackageGit(generators, pkgs) {
+  pkgs = Object.prototype.toString.call(pkgs) === '[object Array]' ? pkgs : [pkgs];
```
Comment (SBoudrias) [inline lib/routes/update.js]: `_.isArray()`

File: lib/routes/update.js [hunk 2]
```diff
@@ -16,8 +16,22 @@ function updateSuccess(app) {
   });
 }
 
+function getPackageGit(generators, pkgs) {
```
Comment (SBoudrias) [inline lib/routes/update.js]: This function name is not very descriptive as to what it's doing.

**Comment by SBoudrias** [comment]
I'm unclear how this is working. Where will the repository data be defined?

**Comment by SBoudrias** [comment]
@landru29 any chance you can address the review comments so we can get the ball rolling on this PR?

**Comment by landru29** [comment]
All your remarks are implemented

**Comment by SBoudrias** [comment]
Hey @landru29, I'm sorry I might've misread the code before and missed that. But right now, this is ALWAYS gonna return the repository url yeah? So that every packages (with a repository key) will be updated from the git repo rather than npm?

**Comment by SBoudrias** [comment]
Given this code only get the git url, this as is is not something we wish to merge in.

If you could differentiate between a module not available on npm and a module that is, then that would be  a better approach.

**Comment by landru29** [comment]
I surrender.
Too difficult to propose you an improvement.
Hard coding "github.com" should not be the right way.

**Pull request closed without merging**